### PR TITLE
Add post-processor to leave only VHD

### LINKED ISSUE #46: Split builder to allow building vm images as well as just vhd's
Create packer-azure-builder-vmimage and packer-azure-builder-vhd. The current behavior of packer-azure-builder is that of packer-azure-builder-vmimage

### PR DESCRIPTION
**Pull request by paulmey**
Fixes #46

### CODE REVIEW

File: packer/builder/azure/artifact_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Op	en Technologies, Inc.
```
Comment (boumenot) [inline packer/builder/azure/artifact_test.go]: Copyright

File: packer/builder/azure/artifact_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		subscriptionID:      "subscriptionID",
+	}
+
+	c.Check(a.State("publishSettingsPath").(string), Equals, "publishSettingsPath")
+	c.Check(a.State("subscriptionID").(string), Equals, "subscriptionID")
+}
+
+func (s *ArtifactSuite) Test_BuilderId(c *C) {
+	a := artifact{}
+	c.Check(a.BuilderId(), Equals, "MSOpenTech.azure")
```
Comment (boumenot) [inline packer/builder/azure/artifact_test.go]: Update.

File: packer/builder/azure/smapi/artifact.go [hunk 3]
```diff
@@ -4,7 +4,11 @@
...
 
-import "fmt"
+import (
+	"log"
+
+	"fmt"
+)
 
 // This is the common builder ID to all of these artifacts.
 const BuilderId = "MSOpenTech.azure"
```
Comment (boumenot) [inline packer/builder/azure/smapi/artifact.go]: To be merged?

File: packer/plugin/packer-post-processor-azure-asm-vhd/main.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
```
Comment (boumenot) [inline packer/plugin/packer-post-processor-azure-asm-vhd/main.go]: Copyright.

File: packer/plugin/packer-post-processor-azure-asm-vhd/main.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
```
Comment (boumenot) [inline packer/plugin/packer-post-processor-azure-asm-vhd/main.go]: Copyright.

File: packer/post-processor/azure-vhd/post-processor.go [hunk 6]
```diff
@@ -0,0 +1,119 @@
+package azurevhd
```
Comment (boumenot) [inline packer/post-processor/azure-vhd/post-processor.go]: Add copyright.

File: packer/post-processor/azure-vhd/post-processor_test.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
+package azurevhd
```
Comment (boumenot) [inline packer/post-processor/azure-vhd/post-processor_test.go]: Copyright.

**Comment by boumenot** [comment]
LGTM :+1:

File: packer/post-processor/azure-vhd/post-processor.go (hunk 6 above)
Comment (paulmey) [inline packer/post-processor/azure-vhd/post-processor.go]: done

File: packer/post-processor/azure-vhd/post-processor_test.go (hunk 7 above)
Comment (paulmey) [inline packer/post-processor/azure-vhd/post-processor_test.go]: done

**Pull request merged**